### PR TITLE
enable test_sampled_addmm_zero_sized_cuda for rocm

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
Enable test_sampled_addmm_zero_sized_cuda_* only for ROCm and CUDA issue is currently active.
Passed since ROCm 5.6

test_sampled_addmm_zero_sized_cuda_float32
test_sampled_addmm_zero_sized_cuda_float64
test_sampled_addmm_zero_sized_cuda_complex64
test_sampled_addmm_zero_sized_cuda_complex128

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

File: test/test_sparse_csr.py [hunk 1]
```diff
@@ -2469,9 +2469,8 @@ def test_sampled_addmm_autograd(self, device, dtype):
             self.assertEqual(a.grad, a1.grad)
             self.assertEqual(b.grad, b1.grad)
 
-    @skipCUDAIfRocm
     @onlyCUDA
-    @skipCUDAIf(True, "Causes CUDA memory exception, see https://github.com/pytorch/pytorch/issues/72177")
+    @skipCUDAIf(not TEST_WITH_ROCM, "Causes CUDA memory exception, see https://github.com/pytorch/pytorch/issues/72177")
```
Comment (pruthvistony) [inline test/test_sparse_csr.py:2473]: Please update the comment to indicate - it works on ROCm and CUDA issue is currently active.

**Review by pruthvistony — CHANGES_REQUESTED** [review]

**Review by jeffdaily — APPROVED** [review]

**Comment by dnikolaev-amd** [comment]
Enabled tests
```xml
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_complex128" time="0.002" file="test_sparse_csr.py"/>
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_complex64" time="0.002" file="test_sparse_csr.py"/>
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_float32" time="0.002" file="test_sparse_csr.py"/>
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_float64" time="0.002" file="test_sparse_csr.py"/>
```

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@malfet 
Can you please review this PR.

**Comment by jithunnair-amd** [comment]
@malfet Please review. Few lines of change to enable some UTs

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**